File: frontend/desktop/src/types/skills.ts
/** Skill record managed by the local skills catalogue. */
export interface Skill {
  id: string;
  name: string;
  description: string;
  instructions: string;
  tags: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

/** Payload for creating a skill. */
export interface CreateSkillPayload {
  name: string;
  description: string;
  instructions: string;
  tags?: string;
  isActive?: boolean;
}

/** Payload for updating a skill. */
export interface UpdateSkillPayload {
  id: string;
  name?: string;
  description?: string;
  instructions?: string;
  tags?: string;
  isActive?: boolean;
}

File: frontend/desktop/src-tauri/src/models/skills.rs
use serde::{Deserialize, Serialize};

/// Frontend-facing skill record.
#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
#[allow(dead_code)]
pub struct Skill {
    pub id: String,
    pub name: String,
    pub description: String,
    pub instructions: String,
    pub tags: String,
    pub is_active: bool,
    pub created_at: String,
    pub updated_at: String,
}
